tests/allschemes/channels007/Main.idr

```diff
@@ -0,0 +1,23 @@
+import System
+import System.Concurrency
+
+-- Test that using channelGetNonBlocking works as expected.
+main : IO ()
+main = do
+  chan <- makeChannel
+  threadID <- fork $ do
+    channelPut chan "Hello"
+    channelPut chan "Goodbye"
+  sleep 1
+  case !(channelGetNonBlocking chan) of
+    Nothing   =>
+      putStrLn "Nothing"
+    Just val' =>
+      putStrLn val'
+  case !(channelGetNonBlocking chan) of
+    Nothing   =>
+      putStrLn "Nothing"
+    Just val' =>
+      putStrLn val'
+  sleep 1
+
```

Comment:
Formatting nits: don't need to split these cases over multiple lines.

```
Nothing => putStrLn "Nothing"
Just val   => putStrLn val
```

This entire case statement can be collapsed to  `putStrLn $ fromMaybe "Nothing" !(channelGetNonBlocking chan)`

or

`putStrLn $ show !(channelGetNonBlocking chan)`, with the caveat that the output will be pretty-printed, but that's probably fine here so long as you adjust your expected values to match

Reply:
This was addressed via commit 6a5129ffafc3537450199fb42a5d7b366f495df1.